test/test_translator.py

```diff
@@ -195,110 +354,253 @@ def test_get_road_direction():
+    'creationtime': '2020-12-13T14:18:00Z', 
+    'description': 'Road constructions are going on',
+    'location': {
+      'polyline': '34.8380671,-114.1450650,34.8380671,-114.1450650',
+      'street': 'I-70 N'
+    }
+  }]}}
+  
+#[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z 
+# [0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}
+
+  wzdx_re='{"road_event_feed_info": {"feed_info_id": "104d7746-688c-44ed-b195-2ee948bf9dfa", "update_date": "[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z", "publisher": "CDOT", "contact_name": "Abinash Konersman", "contact_email": "abinash\\.konersman@state\\.co\\.us", "version": "3\\.0", "data_sources": \\[{"data_source_id": "[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}", "feed_info_id": "104d7746-688c-44ed-b195-2ee948bf9dfa", "organization_name": "iCone", "contact_name": "Abinash Konersman", "contact_email": "abinash\\.konersman@state\\.co\\.us", "update_date": "[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z", "location_method": "channel-device-method", "lrs_type": "lrs_type"}\\]}, "type": "FeatureCollection", "features": \\[{"type": "Feature", "properties": {"road_event_id": "[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}", "event_type": "work-zone", "data_source_id": "[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}", "start_date": "2020-12-07T14:18:00Z", "end_date": "", "start_date_accuracy": "estimated", "end_date_accuracy": "estimated", "beginning_accuracy": "estimated", "ending_accuracy": "estimated", "road_name": "I-70 N", "direction": "northbound", "vehicle_impact": "all-lanes-open", "relationship": {"relationship_id": "[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}", "road_event_id": "[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}"}, "lanes": \\[\\], "road_number": "", "beginning_cross_street": "", "ending_cross_street": "", "event_status": "active", "types_of_work": \\[\\], "reduced_speed_limit": 25, "workers_present": false, "restrictions": \\[\\], "description": "Road constructions are going on", "creation_date": "2020-12-13T14:18:00Z", "update_date": "2020-12-16T17:18:00Z"}, "geometry": {"type": "LineString", "coordinates": \\[\\[-114\\.145065, 34\\.8380671\\], \\[-114\\.145065, 34\\.8380671\\]\\]}}\\]}'
+  test_wzdx = icone_translator.wzdx_creator(icone_obj)
+  assert re.match(wzdx_re,json.dumps(test_wzdx)) != None
+
+def test_wzdx_creator_no_incidents() :
+  icone_obj = {'incidents': {'@timestamp': '2020-12-16T17:18:00Z'}}
+  test_wzdx = icone_translator.wzdx_creator(icone_obj)
+  assert test_wzdx == None
+  
+def test_wzdx_creator_invalid_incidents_no_description() :
+  icone_obj = {'incidents': {'incident': [{
+    '@id': 'U13631595_202012160845',
+    'updatetime': '2020-12-16T17:18:00Z',
+    'starttime': '2020-12-07T14:18:00Z',
+    'creationtime': '2020-12-13T14:18:00Z', 
+    'location': {
+      'polyline': '34.8380671,-114.1450650,34.8380671,-114.1450650',
+      'street': 'I-70 N'
```

Comment:
Can we add to these tests to show that we can have a set of data with both good and bad incidents and we still get the valid incidents back?

Reply:
I added an another test case where we are checking both valid and invalid incidents and it returns wzdx message with only the valid incidents.